src/pymor/vectorarrays/list.py

```diff
@@ -227,10 +232,23 @@ class ListVectorArray(VectorArrayInterface):
 
     _NONE = ()
 
+    _dtype_invalid = True  # force setting dtype on first access
+
+    @invalidates_dtype
     def __init__(self, vectors, space):
         self._list = vectors
         self.space = space
 
+    @property
+    def dtype(self):
+        if self._dtype_invalid:
+            if len(self._list): # else defaults to base class
+                self._cached_dtype = reduce(np.promote_types, (v.dtype for v in self._list))
+            else:
+                self._cached_dtype = VectorArrayInterface.dtype
+            self._dtype_invalid = False
+        return self._cached_dtype
```

Comment:
For `ListVectorArray` we also need to make sure that `dot` and `dofs` return arrays of the right dtype.

Reply:
We also need to make sure that the `dtype` of `ListVectorArrayView` agrees with the `dtype` of the original array. (I think that indexing an array should not change its `dtype`.)